dev-libs/libdnsres/files/libdnsres-0.1a-C99.patch

```diff
@@ -0,0 +1,100 @@
+diff -ru libdnsres-0.1a.orig/arc4random.c libdnsres-0.1a/arc4random.c
+--- libdnsres-0.1a.orig/arc4random.c	2024-04-06 11:39:56.913115613 +0000
++++ libdnsres-0.1a/arc4random.c	2024-04-06 11:44:07.785225323 +0000
+@@ -1,5 +1,6 @@
+ #include <sys/types.h>
+ #include <stdlib.h>
++#include <time.h>
+ 
+ #include "config.h"
+ 
+@@ -9,8 +10,8 @@
+  * application anyway.  Screw you, hippy!
+  */
+ 
+-u_int32_t
+-arc4random(void)
++u_int32_t arc4random(void);
```

Comment:
Let's use C99 `<stdint.h>` types, please.

So, e.g.
```suggestion
+uint32_t arc4random(void);
```

Reply:
They are `<sys/types.h>` types and used everywhere in the program. And appear to be aliaces for uints et all:
```
/* These were defined by ISO C without the first `_'.  */
typedef __uint8_t u_int8_t;
typedef __uint16_t u_int16_t;
typedef __uint32_t u_int32_t;
typedef __uint64_t u_int64_t;
```
and in `<bits/stdint-uintn.h>` included transitively by `<stdint.h>`
```
typedef __uint8_t uint8_t;
typedef __uint16_t uint16_t;
typedef __uint32_t uint32_t;
typedef __uint64_t uint64_t;
```

They are used 75 times in the code and it feels like too much hassle to nuke them off configure and from code.
Maybe later.

Reply:
They're not available on musl, though. Grep `musl.git`.

Reply:
https://git.musl-libc.org/cgit/musl/tree/include/sys/types.h#n62
they pretty much are.

Reply:
Build fails on musl, but that's because inet_nsap_ntoa is missing, not because types are non-existent.